src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,62 @@
+import { Link, useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { getSearchWith } from '../utils/searchHelper';
+import { PersonSex } from '../types/PersonSex';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const query = searchParams.get('query') || '';
+  const sexFilter = searchParams.get('sex') || null;
+  const centuriesFilter = searchParams.getAll('century') || [];
+
+  const centuries = ['16', '17', '18', '19', '20'];
+
+  function handleQueryChange(
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) {
+    setSearchParams(
+      getSearchWith(searchParams, {
+        query: event.target.value || null,
+      }),
+    );
+  }
+
+  const toggleCentury = (century: string) => {
+    return centuriesFilter.includes(century)
+      ? centuriesFilter.filter(cent => cent !== century)
+      : [...centuriesFilter, century];
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+        <Link
+          className={cn({ 'is-active': !sexFilter })}
+          to={{ search: getSearchWith(searchParams, { sex: null }) }}
+        >
+          All
+        </Link>
+
+        <Link
+          className={cn({ 'is-active': sexFilter === PersonSex.male })}
+          to={{ search: getSearchWith(searchParams, { sex: PersonSex.male }) }}
+        >
+          Male
+        </Link>
+
+        <Link
+          className={cn({ 'is-active': sexFilter === PersonSex.female })}
+          to={{
+            search: getSearchWith(
+              searchParams, { sex: PersonSex.female },
+            ),
+          }}
+        >
+          Female
+        </Link>
```

Comment:
You have special component for that in SearchLink.tsx

Reply:
> You have special component for that in SearchLink.tsx

Good point, why you don't use it?

Also, you repeat every time `to={{ search: getSearchWith(searchParams, { sex: PersonSex.male }) }}` only sex changes, you create a function in this case